RocketChat/Rocket.Chat.ReactNative
# [FIX] Clear settings on server change

@RocketChat/ReactNative


This PR fixes some settings getting dirty after changing to a server with a previous version.

### Test plan 1
- Log into open server (version 3.1.0)
  - Check `DirectMesssage_maxUsers` setting on redux
- Log into stable server (version 2.3.2)
  - `DirectMesssage_maxUsers` shouldn't exist on redux

### Test plan 2
- Log into open server (version 3.1.0)
- Add a server with different `Accounts_ShowFormLogin` or `Accounts_RegistrationForm`
  - The app should have changed those settings accordingly
